backend/server/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java

```diff
@@ -205,7 +205,9 @@ public InternalFuture<List<KeyValue>> getAttributes(GetAttributes request) {
             unused ->
                 checkProjectPermission(projectId, ModelDBActionEnum.ModelDBServiceActions.READ),
             executor)
-        .thenCompose(unused -> attributeHandler.getKeyValues(projectId, keys, getAll), executor);
+        .thenCompose(
```

Comment:
Could this be `thenSupply()`?

Reply:
definitely. I stopped trying to fix everything I saw after a while. :)